plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1281,6 +1290,10 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
       } );
 
       my->irreversible_block_connection = my->chain->irreversible_block.connect( [this]( const block_state_ptr& blk ) {
+         if (my->chain_config->stop_at_lib > 0 && blk->block_num >= my->chain_config->stop_at_lib) {
+            wlog( "Stopping at lib#${num}.", ("num", blk->block_num));
+            appbase::app().quit();
```

Comment:
This simply signals application to shutdown. All queued posted jobs to the main thread complete which could potentially include some more blocks.